### ISSUE TITLE
Broadcasting Operations

### ISSUE THREAD

**Issue by oxinabox** [invenia/NamedDims.jl#4]
Needs to be done following similar rules to #3  
and further:
- unitary operations keep dim names
- `NamedDimArray` op scalar: keeps dim names.

**Comment by mcabbott** [comment]
I was about to open an issue, but perhaps this isn't expected to work yet... anyway this broadcasting operation gives an error:
```
julia> nda = NamedDimsArray{(:x,:y,:z)}(rand(10,20,30)); # as in readme, but Array not Arrays

julia> nda .+ rand(20)'
ERROR: duplicate field name in NamedTuple: "_" is not unique
Stacktrace:
 [1] identity_namedtuple(::Tuple{Symbol,Symbol}) at /Users/me/.julia/packages/NamedDims/JWdM2/src/name_core.jl:78
 [2] combine_names_longest(::Tuple{Symbol,Symbol,Symbol}, ::Tuple{Symbol,Symbol}) at /Users/me/.julia/packages/NamedDims/JWdM2/src/name_core.jl:201
 [3] broadcasted_names at /Users/me/.julia/packages/NamedDims/JWdM2/src/broadcasting.jl:20 [inlined]
 ```
Curiously it doesn't happen for 3 dimensions:
```
julia> nda .+ rand(1,1,30);

julia> nda .+ rand(10);

julia> nda .+ rand(1,20);
ERROR: duplicate field name in NamedTuple: "_" is not unique
```

**Comment by oxinabox** [comment]
This issue should have been closed by #9 

Your particular issue @mcabbott is basically part of #8

**Comment by mcabbott** [comment]
OK, I wondered if that was related, thanks for taking a look!
